kratos/CMakeLists.txt

```diff
@@ -222,6 +223,7 @@ if(USE_COTIRE MATCHES ON)
     ## Kratos cannot be compiled with just one unity build as it takes way too much ram
     SET(COTIRE_MAXIMUM_NUMBER_OF_UNITY_INCLUDES "-j2")
     set_source_files_properties (${CMAKE_CURRENT_SOURCE_DIR}/sources/kratos_parameters.cpp PROPERTIES COTIRE_EXCLUDED TRUE)
+    set_source_files_properties (${CMAKE_CURRENT_SOURCE_DIR}/sources/memory_info.cpp PROPERTIES COTIRE_EXCLUDED TRUE)
```

Comment:
I don't know why this is necessary, but win crashes without :/

Reply:
Again, its the minmax problem of `windows.h`, only that this time happens because it overwrites it in the boundingbox code:

```c++
for (auto& r_node : rGeometry) {
        const array_1d<double, 3> vector_points = r_node.Coordinates() - mPointCenter;
        distance_0 = std::max(distance_0, std::abs(inner_prod(vector_points, mOrientationVectors[0])));
        distance_1 = std::max(distance_1, std::abs(inner_prod(vector_points, mOrientationVectors[1])));
        distance_2 = std::max(distance_2, std::abs(inner_prod(vector_points, mOrientationVectors[2])));
    }
```

Warding the include with the `#define NOMINMAX` should solve it

Reply:
hm did not solve it

Reply:
No idea then :S I will try to take a look at home (have no usable windows for that with me right now)

Reply:
Ho I found it, you are gonna love it: 
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getmessage

Reply:
Also, `#undef GetMessage` after the include should solve the issue, provided that we are not using that anyway.

Reply:
Aslo `#define INTERFACE   IPrintDialogServices` from `commdlg.h` I will take it as a joke an laugh.

Reply:
> .

I am scared to look at this ...

Reply:
bth your last commit didn't fix it